.github/workflows/test-platforms.yml

```diff
@@ -7,21 +7,63 @@ on:
     types: [opened, synchronize, reopened]
 
 jobs:
-  Build:
+  job:
+    name: ${{ matrix.os }} - ${{ github.workflow }}
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: windows-latest
+            triplet: x64-windows
+          - os: ubuntu-latest
+            triplet: x64-linux
+          - os: macos-latest
+            triplet: x64-osx
+
+    env:
+      VCPKG_DEFAULT_TRIPLET: ${{ matrix.triplet }}
+
     steps:
       - uses: actions/checkout@v2
-      # build project
-      - name: make build directory
+        with:
+          submodules: true
+
+      - name: Vcpkg bootstrap Windows
+        run: bootstrap-vcpkg.bat
+        working-directory: vcpkg
+        if: ${{ matrix.os == 'windows-latest' }}
+
+      - name: Vcpkg bootstrap Shell
+        run: ./bootstrap-vcpkg.sh
+        shell: bash
+        working-directory: vcpkg
+        if: ${{ matrix.os != 'windows-latest' }}
+
+      - name: Mac Install Brew
+        run: brew install automake libtool
+        if: ${{ matrix.os == 'macos-latest' }}
+
+      - name: Install ICU
+        run: vcpkg install
+        working-directory: vcpkg
+
+      - name: Make build directory
         run: mkdir build
-      - name: Cmake Configure
-        run: cmake -Bbuild -S.
-      - name: Cmake Build all targets on Windows
-        run: cmake --build build/ --target ALL_BUILD
+
+      - name: Generate Solution Windows
+        run: cmake -B build -A Win32 -DCMAKE_TOOLCHAIN_FILE='${{ github.workspace }}/vcpkg/scripts/buildsystems/vcpkg.cmake'
         if: ${{ matrix.os == 'windows-latest' }}
-      - name: Cmake Build all targets on Unix
-        run: cmake --build build/ --target all
+
+      - name: Generate Solution Linux
+        run: cmake -DCMAKE_BUILD_TYPE=Release -Bbuild -S.
```

Comment:
Should this be ` cmake -DCMAKE_BUILD_TYPE=Release -B build -S .`?

Reply:
It worked for Linux. It is a bit strange but some coders like "compact". I'm not partial - either way.